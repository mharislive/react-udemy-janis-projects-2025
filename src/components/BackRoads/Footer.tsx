import Container from "./Container";
import Navigation from "./Navigation";
import Section from "./Section";
import SocialIcons from "./SocialIcons";

const Footer = () => {
  return (
    <footer>
      <Section className="bg-blue-950">
        <Container className="text-white">
          <Navigation className="mb-6" aria-label="footer navigation" />
          <SocialIcons
            label="Footer social icons"
            color="text-white"
            className="justify-center mb-6"
          />
          <div className="capitalize tracking-widest flex justify-center">
            copyright Â© Backroads travel tours company{" "}
            {new Date().getFullYear()} all rights reserved
          </div>
        </Container>
      </Section>
    </footer>
  );
};

export default Footer;
