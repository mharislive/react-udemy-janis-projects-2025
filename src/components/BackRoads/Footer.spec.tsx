import { render, screen, within } from "@testing-library/react";
import { axe } from "vitest-axe";
import Footer from "./Footer";

describe("BackRoads - Footer", () => {
  it("renders navigation links", () => {
    render(<Footer />);
    const nav = screen.getByRole("navigation", { name: /footer navigation/i });
    expect(within(nav).getByText(/Home/i)).toBeInTheDocument();
    expect(within(nav).getByText(/About/i)).toBeInTheDocument();
    expect(within(nav).getByText(/Services/i)).toBeInTheDocument();
    expect(within(nav).getByText(/Tours/i)).toBeInTheDocument();
  });

  it("renders SocialIcons with correct props", () => {
    render(<Footer />);
    const socialIcons = screen.getByLabelText(/footer social icons/i);
    expect(socialIcons).toBeInTheDocument();
  });

  it("renders copyright text with current year", () => {
    render(<Footer />);
    const year = new Date().getFullYear();
    expect(
      screen.getByText(
        new RegExp(
          `copyright Â© Backroads travel tours company ${year} all rights reserved`,
          "i"
        )
      )
    ).toBeInTheDocument();
  });

  it("should match snapshot", () => {
    const { asFragment } = render(<Footer />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("should have no accessibility violations", async () => {
    const { container } = render(<Footer />);
    const results = await axe(container);
    expect(results).toHaveNoViolations();
  });
});
