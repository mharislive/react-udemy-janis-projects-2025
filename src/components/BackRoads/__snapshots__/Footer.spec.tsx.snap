// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BackRoads - Footer > should match snapshot 1`] = `
<DocumentFragment>
  <footer>
    <section
      class="py-20 bg-blue-950"
      id=""
    >
      <div
        class="container max-w-[1210px] mx-auto px-5 text-white"
      >
        <nav
          aria-label="footer navigation"
          class="flex gap-3 justify-center mb-6"
        >
          <a
            class="transition duration-500 hover:text-cyan-500 tracking-widest"
            href="#"
          >
            Home
          </a>
          <a
            class="transition duration-500 hover:text-cyan-500 tracking-widest"
            href="#"
          >
            About
          </a>
          <a
            class="transition duration-500 hover:text-cyan-500 tracking-widest"
            href="#"
          >
            Services
          </a>
          <a
            class="transition duration-500 hover:text-cyan-500 tracking-widest"
            href="#"
          >
            Tours
          </a>
        </nav>
        <nav
          aria-label="Footer social icons"
          class="flex gap-3 justify-center mb-6"
        >
          <a
            aria-label="facebook"
            class="text-white hover:text-cyan-300"
            href="#"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-facebook fa-2x "
              data-icon="facebook"
              data-prefix="fab"
              focusable="false"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M512 256C512 114.6 397.4 0 256 0S0 114.6 0 256C0 376 82.7 476.8 194.2 504.5V334.2H141.4V256h52.8V222.3c0-87.1 39.4-127.5 125-127.5c16.2 0 44.2 3.2 55.7 6.4V172c-6-.6-16.5-1-29.6-1c-42 0-58.2 15.9-58.2 57.2V256h83.6l-14.4 78.2H287V510.1C413.8 494.8 512 386.9 512 256h0z"
                fill="currentColor"
              />
            </svg>
          </a>
          <a
            aria-label="twitter"
            class="text-white hover:text-cyan-300"
            href="#"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-twitter fa-2x "
              data-icon="twitter"
              data-prefix="fab"
              focusable="false"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M459.37 151.716c.325 4.548.325 9.097.325 13.645 0 138.72-105.583 298.558-298.558 298.558-59.452 0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055 0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421 0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391 0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04 0-57.828 46.782-104.934 104.934-104.934 30.213 0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253z"
                fill="currentColor"
              />
            </svg>
          </a>
          <a
            aria-label="squarespace"
            class="text-white hover:text-cyan-300"
            href="#"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-squarespace fa-2x "
              data-icon="squarespace"
              data-prefix="fab"
              focusable="false"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M186.12 343.34c-9.65 9.65-9.65 25.29 0 34.94 9.65 9.65 25.29 9.65 34.94 0L378.24 221.1c19.29-19.29 50.57-19.29 69.86 0s19.29 50.57 0 69.86L293.95 445.1c19.27 19.29 50.53 19.31 69.82.04l.04-.04 119.25-119.24c38.59-38.59 38.59-101.14 0-139.72-38.59-38.59-101.15-38.59-139.72 0l-157.22 157.2zm244.53-104.8c-9.65-9.65-25.29-9.65-34.93 0l-157.2 157.18c-19.27 19.29-50.53 19.31-69.82.05l-.05-.05c-9.64-9.64-25.27-9.65-34.92-.01l-.01.01c-9.65 9.64-9.66 25.28-.02 34.93l.02.02c38.58 38.57 101.14 38.57 139.72 0l157.2-157.2c9.65-9.65 9.65-25.29.01-34.93zm-261.99 87.33l157.18-157.18c9.64-9.65 9.64-25.29 0-34.94-9.64-9.64-25.27-9.64-34.91 0L133.72 290.93c-19.28 19.29-50.56 19.3-69.85.01l-.01-.01c-19.29-19.28-19.31-50.54-.03-69.84l.03-.03L218.03 66.89c-19.28-19.29-50.55-19.3-69.85-.02l-.02.02L28.93 186.14c-38.58 38.59-38.58 101.14 0 139.72 38.6 38.59 101.13 38.59 139.73.01zm-87.33-52.4c9.64 9.64 25.27 9.64 34.91 0l157.21-157.19c19.28-19.29 50.55-19.3 69.84-.02l.02.02c9.65 9.65 25.29 9.65 34.93 0 9.65-9.65 9.65-25.29 0-34.93-38.59-38.59-101.13-38.59-139.72 0L81.33 238.54c-9.65 9.64-9.65 25.28-.01 34.93h.01z"
                fill="currentColor"
              />
            </svg>
          </a>
        </nav>
        <div
          class="capitalize tracking-widest flex justify-center"
        >
          copyright Â© Backroads travel tours company 2025 all rights reserved
        </div>
      </div>
    </section>
  </footer>
</DocumentFragment>
`;
